pkg/uki/upgrade.go

```diff
@@ -74,3 +82,37 @@ func (i *UpgradeAction) Run() (err error) {
 
 	return nil
 }
+
+// installRecovery replaces the "recovery" role efi and conf files with
+// the UnassignedArtifactRole efi and loader files from dir
+func (i *UpgradeAction) installRecovery(logger v1.Logger) error {
```

Comment:
Why passing a logger when there is one at i.cfg.logger?

Reply:
oops good catch! Actually that's the logger I'm passing to the method. It was like that because initially I didn't make this a method of UpgradeAction (it had no receiver). I then moved it under `UpgradeAction` but didnt' notice I no longer needed to pass the logger. Fixing.